frontend/src/container/LogsExplorerViews/index.tsx

```diff
@@ -318,6 +336,23 @@ function LogsExplorerViews({
 		],
 	);
 
+	useEffect(() => {
+		setQueryId(Math.random().toString(36).slice(2));
+	}, [isError, isSuccess]);
+
+	useEffect(() => {
+		if (
+			!isEmpty(queryId) &&
+			(isLoading || isFetching) &&
+			selectedPanelType !== PANEL_TYPES.LIST
+		) {
+			setQueryStats(undefined);
+			setTimeout(() => {
+				getQueryStats({ queryId, setData: setQueryStats });
+			}, 500);
```

Comment:
Reason for delay ?

Reply:
there is a delay between the `query_range` API reaching the query service and ws connection, if the connection is being tried to establish before the query_range API reaching the qs it fails hence the delay.